src/transformers/processing_utils.py

```diff
@@ -409,6 +607,111 @@ def from_args_and_dict(cls, args, processor_dict: Dict[str, Any], **kwargs):
+                image_processor = image_processor_class(...)
+                processor(tokenizer, image_processor) # will pass max_length unless overriden by kwargs at call
+                ```
+            5) defaults kwargs specified at processor level have lowest priority.
+
+        Args:
+            ModelProcessorKwargs (`ProcessingKwargs`):
+                Typed dictionary of kwargs specifically required by the model passed.
+            text_kwargs (`TextKwargs`, *optional*):
+                Typed dictionary of kwargs inputs applied to the text modality processor, i.e. the tokenizer.
+            images_kwargs (`ImagesKwargs`, *optional*):
+                Typed dictionary of kwargs inputs applied to the images modality processor.
+            videos_kwargs (`VideosKwargs`, *optional*):
+                Typed dictionary of kwargs inputs applied to the videos modality processor.
+            audio_kwargs (`AudioKwargs`, *optional*):
+                Typed dictionary of kwargs inputs applied to the audio modality processor.
+            tokenizer_init_kwargs (`Dict`, *optional*):
+                Dictionary of kwargs the tokenizer was instantiated with and need to take precedence over other kwargs.
+
+        Returns:
+            output_kwargs (`Dict`):
+                Dictionary of per-modality kwargs to be passed to each modality-specific processor.
+
+        """
+
+        # Initialize dictionaries
+        output_kwargs = {
+            "text_kwargs": text_kwargs or {},
+            "images_kwargs": images_kwargs or {},
+            "audio_kwargs": audio_kwargs or {},
+            "videos_kwargs": videos_kwargs or {},
+            "common_kwargs": common_kwargs or {},
+        }
+
+        default_kwargs = {
+            "text_kwargs": {},
+            "images_kwargs": {},
+            "audio_kwargs": {},
+            "videos_kwargs": {},
+            "common_kwargs": {},
+        }
+
+        # get defaults from set model processor kwargs if they exist
+        for modality in default_kwargs:
+            default_kwargs[modality] = ModelProcessorKwargs._defaults.get(modality, {}).copy()
+        # then override with tokenizer-level arguments passed
+        if tokenizer_init_kwargs:
+            default_kwargs["text_kwargs"].update(
+                {k: v for k, v in tokenizer_init_kwargs.items() if k in ModelProcessorKwargs.text_kwargs}
+            )
+
+        # then get passed per-modality dictionaries if they exist
+        for modality in output_kwargs:
+            output_kwargs[modality] = {
+                **default_kwargs[modality],
+                **output_kwargs[modality],
+                **kwargs.pop(modality, {}),
+            }
+            # then merge kwargs by name
+            for modality_key in ModelProcessorKwargs[modality].__annotations__.keys():
+                modality_kwarg_value = kwargs.pop(modality_key, None)
+                if modality_kwarg_value is not None:
```

Comment:
Is there any case we might want to pass `None`?

Reply:
I mean, let`s suppose we pass

```python
inputs = processor(image, text, size_divisor=None)  # suppose it defaults to `32` and we want to turn it off
```

In such a case, I guess, kwargs will not be updated. The example is a bit 

I suppose it would be more safe:
```
if modality_key in kwargs:
  output_kwargs[modality] =  kwargs.pop(modality_key)
```

Reply:
It's true we might want to turn off some of them - will add your suggestion!

Reply:
I think I'll add these use cases (preserve defaults, pass correctly kwargs, in short check that priority order is respected) to a common testing util in order to avoid regressions on that - kwargs are a bit mind bending

Reply:
> kwargs are a bit mind bending
absolutely 😁💯